libs/core/tests/unit_tests/runnables/test_context.py

```diff
@@ -0,0 +1,315 @@
+from typing import Any, Callable, List, NamedTuple, Union
+
+import pytest
+
+from langchain_core.output_parsers.string import StrOutputParser
+from langchain_core.prompt_values import StringPromptValue
+from langchain_core.prompts.prompt import PromptTemplate
+from langchain_core.runnables.base import Runnable, RunnableLambda
+from langchain_core.runnables.context import Context
+from langchain_core.runnables.passthrough import RunnablePassthrough
+from langchain_core.runnables.utils import aadd, add
+from tests.unit_tests.fake.llm import FakeListLLM, FakeStreamingListLLM
+
+
+class TestCase(NamedTuple):
+    input: Any
+    output: Any
+
+
+def seq_naive_rag() -> Runnable:
+    context = [
+        "Hi there!",
+        "How are you?",
+        "What's your name?",
+    ]
+
+    retriever = RunnableLambda(lambda x: context)
+    prompt = PromptTemplate.from_template("{context} {question}")
+    llm = FakeListLLM(responses=["hello"])
+
+    return (
+        Context.setter("input")
+        | {
+            "context": retriever | Context.setter("context"),
+            "question": RunnablePassthrough(),
+        }
+        | prompt
+        | llm
+        | StrOutputParser()
+        | {
+            "result": RunnablePassthrough(),
+            "context": Context.getter("context"),
+            "input": Context.getter("input"),
+        }
+    )
+
+
+def seq_naive_rag_alt() -> Runnable:
+    context = [
+        "Hi there!",
+        "How are you?",
+        "What's your name?",
+    ]
+
+    retriever = RunnableLambda(lambda x: context)
+    prompt = PromptTemplate.from_template("{context} {question}")
+    llm = FakeListLLM(responses=["hello"])
+
+    return (
+        Context.setter("input")
+        | {
+            "context": retriever | Context.setter("context"),
+            "question": RunnablePassthrough(),
+        }
+        | prompt
+        | llm
+        | StrOutputParser()
+        | Context.setter("result")
+        | Context.getter(["context", "input", "result"])
+    )
+
+
+test_cases = [
```

Comment:
i dont think you have a test case of `Context.setter(foo=lambdax)` (eg with no key provided)

Reply:
added